tests/test_inputcapture.py

```diff
@@ -59,53 +64,63 @@ def create_session(self, portal_mock, capabilities=0xF):
         self.current_session_handle = results["session_handle"]
 
         # Check the impl portal was called with the right args
-        method_calls = portal_mock.mock_interface.GetMethodCalls("CreateSession")
+        method_calls = mock_intf.GetMethodCalls("CreateSession")
         assert len(method_calls) > 0
         _, args = method_calls[-1]
         assert args[3] == ""  # parent window
         assert args[4]["capabilities"] == capabilities
 
         return response, results
 
-    def get_zones(self, portal_mock):
+    def get_zones(self, dbus_con):
         """
         Call GetZones and return the (response, results) tuple.
         """
-        request = portal_mock.create_request()
-        options = {}
-        response, results = request.call(
+        inputcapture_intf = xdp.get_portal_iface(dbus_con, "InputCapture")
+        mock_intf = xdp.get_mock_iface(dbus_con)
+
+        request = xdp.Request(dbus_con, inputcapture_intf)
+        options: Any = {}
+        res = request.call(
```

Comment:
s/res/ret/ (in a few places)

Reply:
Changed it to the common pattern of calling the returned tuple `response` and accessing `response.response` or `response.results`.